generators/server/templates/src/main/java/package/repository/_UserRepository.java

```diff
@@ -41,9 +39,11 @@
     Optional<User> findOneByLogin(String login);
     <%_ if (databaseType == 'sql') { _%>
 
-    @Query(value = "select distinct user from User user left join fetch user.authorities",
-        countQuery = "select count(user) from User user")
-    Page<User> findAllWithAuthorities(Pageable pageable);
+    @EntityGraph(attributePaths = "authorities")
+    User findOneWithAuthoritiesById(<%= pkType %> id);
+
+    @EntityGraph(attributePaths = "authorities")
+    Optional<User> findOneWithAuthoritiesByLogin(String login);
```

Comment:
For this, query cache need to be activated because the request is not made on the primary key
`hibernate.cache.use_query_cache: true`

and enable it for the query in the repository
`@QueryHints({@QueryHint(name = "org.hibernate.cacheable", value = "true")})
Optional<User> findOneByLogin(String login);`